src/simtools/simulator.py

```diff
@@ -786,3 +761,111 @@ def _get_calibration_device_types(run_mode):
         if run_mode == "direct_injection":
             return ["flat_fielding"]
         return []
+
+    def verify_simulations(self):
+        """
+        Verify simulations.
+
+        This includes checking the number of simulated events.
+
+        """
+        self.logger.info("Verifying simulations.")
+
+        expected_shower_events = self.args_dict.get("nshow", 0)
+        # core scatter is a list: first element is the usage factor
+        expected_mc_events = expected_shower_events * self.args_dict.get("core_scatter", [0])[0]
+
+        self.logger.info(
+            f"Expected number of shower events: {expected_shower_events}, "
+            f"expected number of MC events: {expected_mc_events}"
+        )
+
+        if self.simulation_software in ["corsika_sim_telarray", "sim_telarray"]:
+            self._verify_simulated_events_in_sim_telarray(
+                expected_shower_events, expected_mc_events
+            )
+        if self.args_dict.get("save_reduced_event_lists"):
+            self._verify_simulated_events_in_reduced_event_lists(expected_mc_events)
```

Comment:
Do I understand correctly that if the reduced event lists is saved, we count the number of events in the file twice? It might not cost that much time (to be seen), but is probably redundant, no? I think that if we save the reduced event lists, it's enough check only with `_verify_simulated_events_in_reduced_event_lists`

Reply:
Yes - do we really couple writing of event lists and the testing of sim_telarray files? I think they should be logical decoupled and the overhead of counting objects in a file should not be large.